src/app/layout.tsx

```diff
@@ -1,43 +1,64 @@
-import { AppRouterCacheProvider } from '@mui/material-nextjs/v14-appRouter';
-import { ThemeProvider, createTheme } from '@mui/material/styles';
-import type { Metadata } from "next";
-import { ToastContainer } from 'react-toastify';
-import 'react-toastify/dist/ReactToastify.css';
+"use client";
```

Comment:
always don't make the layout of the whole app client component
there is another solution to make drawer in all pages except sign in and sign out without make this layout client 
so think of it, and tomorrow before standup meeting or after it , we will talk about it.